cpp/daal/src/algorithms/low_order_moments/oneapi/low_order_moments_distributed_oneapi_impl.i

```diff
@@ -0,0 +1,488 @@
+/* file: low_order_moments_distributed_oneapi_impl.i */
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+/*
+//++
+//  Low order moments algorithm implementation in distributed mode.
+//--
+*/
+
+#ifndef __LOW_ORDER_MOMENTS_DISTRIBUTED_ONEAPI_IMPL_I__
+#define __LOW_ORDER_MOMENTS_DISTRIBUTED_ONEAPI_IMPL_I__
+
+#include "services/internal/buffer.h"
+#include "data_management/data/numeric_table.h"
+#include "services/env_detect.h"
+#include "services/error_indexes.h"
+#include "src/algorithms/low_order_moments/oneapi/cl_kernels/low_order_moments_kernels_distr.h"
+#include "src/algorithms/low_order_moments/oneapi/low_order_moments_kernel_distributed_oneapi.h"
+#include "src/externals/service_ittnotify.h"
+#include "services/internal/execution_context.h"
+#include "services/daal_defines.h"
+
+using namespace daal::services::internal;
+using namespace daal::services::internal::sycl;
+
+namespace daal
+{
+namespace algorithms
+{
+namespace low_order_moments
+{
+namespace oneapi
+{
+namespace internal
+{
+#define RET_IF_FAIL(st) \
+    if (!st)            \
+    {                   \
+        return;         \
+    }
+#define CHECK_AND_RET_IF_FAIL(st, expr) \
+    (st) |= (expr);                     \
+    if (!st)                            \
+    {                                   \
+        return;                         \
+    }
```

Comment:
Please check [this](https://github.com/oneapi-src/oneDAL/blob/0295041bd405f68e150558793a28193e5a065369/cpp/daal/include/services/daal_defines.h#L483). Such
 functionality has been already implemented.

Reply:
thanks, both of them removed